### DISCUSSION TITLE
Motion correction

### DISCUSSION THREAD

**Discussion by ProjX0** [llnl/LEAP#63] [Ideas]
Hello,

I was wondering if there are any plans to add motion correction features to the LEAP toolkit.

If so, I have attached two interesting papers(links) that could be valuable additions.

Integrating these motion correction capabilities into the LEAP toolkit could be extremely beneficial.

Thank you.

Best regards,

https://pubmed.ncbi.nlm.nih.gov/25979048/
https://ieeexplore.ieee.org/document/7820191

**Comment by scf819** [comment]
It seems to be similar to Parker weigth~ DOI: 10.1118/1.1450132
Some angles have been overlooked?

**Reply by ProjX0** [reply]
Oh, this looks like a very interesting paper. I haven't had a chance to read it yet, but I will definitely take a look. Thank you.

**Comment by kylechampley** [comment]
Ya, I've seen algorithms like that.  I worked on similar things when I worked in the CT group at GE.

I have no plans to add such an algorithm like that to LEAP, but you could use LEAP as a library to implement those algorithms yourself.

I do take requests from users when adding new features to LEAP.  If enough people want a certain feature and it isn't too much work, I'll do it.

**Comment by ProjX0** [comment]
It's truly impressive that you have done similar work at GE. In my opinion, the LEAP toolkit is the best I’ve encountered so far. I believe that motion correction is something many others would want as well, and its inclusion could attract even more users. It's a shame that there are no plans to add it yet, but I appreciate your consideration. Thank you for always providing such prompt responses.

Does the LEAP library include a function for "cone-parallel geometry via row-wise fan-to-parallel rebinning"?

**Reply by kylechampley** [reply]
A new version of LEAP was just released which includes functionality supporting cone-parallel coordinates.

See geometry specification [here](https://leapct.readthedocs.io/en/latest/ctgeometries.html#leapctype.tomographicModels.set_coneparallel)

And rebinning algorithm [here](https://leapct.readthedocs.io/en/latest/projection_filters_and_transforms.html#leapctype.tomographicModels.rebin_parallel)

**Reply by ProjX0** [reply]
Kyle,

Thank you so much for the update and for providing the links to the new LEAP version with the cone-parallel coordinates functionality.
I truly appreciate the responses and the continuous improvements to the toolkit. I'll definitely give this new feature a try and see how it works.

Thanks again!

**Comment by kylechampley** [comment]
My current job is in SPECT, so my work for LEAP is a hobby of mine that I do in my spare time.  I do plan to continue to add features, but there is a limit in what I can do.

Anyway, what you speak of is commonly referred to as cone-parallel coordinates.  It is actually near the top of my TODO list for LEAP.  Have you seen this list on the LEAP homepage [here](https://github.com/LLNL/LEAP?tab=readme-ov-file#future-releases)?

**Reply by ProjX0** [reply]
Thank you for your response and for the hard work you put into the LEAP toolkit, even in your spare time. I understand the limitations and appreciate your dedication to continuously improving it.
I'm glad to hear that cone-parallel coordinates are already on your TODO list. I have not seen the TODO list on the LEAP homepage yet, but I will definitely check it out. Thank you for pointing it out.

**Comment by hws203** [comment]
I also hope that LEAP can support  motion correction feature. And I think that Leap already supports fast projection by GPU code, so projection based correction algorithm may be easy for  @kylechampley.  So I added another paper to here.
[Revised_Projection_based_dynamic_tomography__PMB.pdf](https://github.com/user-attachments/files/15997196/Revised_Projection_based_dynamic_tomography__PMB.pdf)
